draft-scheffenegger-congress-rfc5033bis.md

```diff
@@ -292,6 +291,39 @@ trip times comparable bandwidth during backoff.
 alternate congestion control algorithm, the proposal should
 explore how bandwidth is shared among the competing flows.
 
+(*)
+: Utilising More than one Path.
+
+: An IETF transport is not tied to a specific Internet path.
+The set of routers forming the path can and do change with time,
+this will also cause the properties of the path to change with respect to time.
+Congestion control needs to be designed to take into account
+these changes in the path characteristics.
+Some transports permit more than one path to be differentiated and used by
+a single connection at the sender. These are called multipath transport protocols.  
+A multipath sender can schedule which packets travel on which of its active paths.
+This enables a tradeoff in timeliness and reliability.
+
+One use is to provide fail-over from one path to
+another when the original path is no longer viable or to switch the traffic from
+one path to another when this is expected to improve performance
+(latency, throughput, reliability, cost).
+A design independently needs to track the congestion state of each path,
+and needs to demonstrate independent congestion control for each path being used.
+
+A concurrent transport protocol simultaneously
+schedules multiple flows to aggregate the capacity of multiple paths.
+This introduces additional implications:
+The Internet provides no guarantee that different paths
+(e.g., using different endpoint addresses) are disjoint,
+therefore a design needs to evaluate the potential
+harm to other flows (including single path connections)
+that share a common bottleneck
+(or share resources that are coupled between different paths,
+such as an overall capacity limit).
```

Comment:
I think this is all too long, and mixes two or three different concepts:

1) that routing change happen, and that congestion control need to be ready for such changes.

2) that transport protocols managing different paths should manage congestion independently for each path.

3) that even when paths are defined by different pairs of addresses, there is a possibility that these paths share some common congested segments.

The first consideration is not really about multipath, and in my mind should be pulled into its own subsection.

There is some consensus about "measuring the congestion for each path", by opposition to "measuring the congestion control for a bundle paths", which could cause congestion on the lowest capacity path of the bundle. This is summarized in TCPM (section 3.3.7 of RFC 8684) as "we mean that it does not take up more capacity on any one path than if it was a single path flow using only that route, so this ensures fair coexistence with single-path TCP at shared bottlenecks." There is a similar point in QUIC (Section 9.4 RFC 9000) and in the QUIC Multipath draft (Section 7.2. of draft-ietf-quic-multipath), "When the QUIC multipath extension is used, senders manage per-path congestion status as required in Section 9.4 of [QUIC-TRANSPORT]."

The third point about common segments is touching into research issues. If the different paths share a common segment, we may want to do something, but what? The easy solution is that if N path share the same segment, the effect on other users must not be worse than if N independent connections shared this segment. Which is of course a low bar. But then, detecting that paths do in fact share segments is a bit of a research issue. Yes, we have a coupled congestion control algorithm specified in RFC 6356, but this is an experimental RFC, which mean we cannot really mandate its use.

Reply:
1. is indeed generic (but p-lease talk about "forwarding" not "routing" - which includes more than a route change, I would be happy to see that appear separately. 
2. I agree is important to say something.
3. I don't agree with your proposal above. I would really like to suggest is still an issue to call-out and see what text people include in future in response.

Reply:
I don't mind calling out the issue, but I don't think we can make specific recommendations.

Reply:
OK, I'll try a PR, and we can always update later, if we need to.

Reply:
We are due for a big re-org here so I would discourage worrying which section text goes into right now.

Furthermore, I think this could benefit from thinking more about test cases and less about advice for algorithm developers. So, for (1) we should definitely evaluate relative performance with path changes.

For (2) I definitely know what you both mean, but I'm having trouble thinking about how to frame this in a usable way. What if a congestion control looked at the relative state of two paths and made a path decision based on that? What if Path A and B share the same bottleneck -- isn't coupling reasonable then? What is the testable assertion here? Perhaps "vary path B  state and show that path A throughput is unchanged," but what are we hoping to gain from that? I am creeping towards the idea that we don't want to say anything about this at all, though I can certainly be persuaded otherwise.

(3) There are zero standard concurrent multipath congestion controls at this time. We may not be able to say anything, but it would certainly be nice if the community could converge on something here. Strict fairness over a  shared bottleneck sits poorly with our theme of moving away from fairness to starvation. So is it OK for an algorithm to have 3 convergent paths and thus take 3/4 of the bandwidth there?

Reply:
To answer Martin's question, yes, it is OK for an algorithm to have 3 convergent paths and thus take 3/4 of the bandwidth. To elaborate a bit:

1) the remaining connection in the 3/4th case has 1/4th of the bandwidth, so it is not "starving".

2) in a lot of cases, we have just one connection sharing the bottleneck, even with N paths. Self interest means deployments will not want to do something silly.

3) if an admin wants to ensure fairness, they should deploy an AQM that enforces it. Relying on the congestion control algorithm for that seems fragile.

Reply:
I think 2 is common, but I am unsure it is helpful as a safety check. It was so hard to know if a flow from one content provider harms that of another. I think this is a more general question though - not just multipath.

Reply:
Thought resolved enough to merge